### ISSUE TITLE
Question on Consistency Model and Diffusion Model Loss

### ISSUE THREAD

**Issue by Messimanda** [xandergos/sCM-mnist#7]
Thanks again for your great work! Regarding the train consistency model part, shouldn't the loss be written like this:
_loss = (torch.exp(logvar) / D) * torch.square(F_theta - F_theta_minus - g) - logvar_

But in your code, it's:
_loss = (weight / torch.exp(logvar)) * torch.square(F_theta - F_theta_minus - g) + logvar_

I made this modification based on Algorithm 1 in the paper. If I misunderstood, I’d appreciate your correction.

Additionally, for the train diffusion part, I still have some confusion about your loss implementation.
Your code uses:
_loss = (1 / torch.exp(logvar)) * torch.square((pred_v_t - v_t) / sigma_data) + logvar_

But according to Equation 25 and Appendix D in the paper, shouldn't it instead be:
_loss = (torch.exp(logvar) / D) * torch.square(pred_v_t - v_t) - logvar_

Thanks for your clarification!

**Comment by xandergos** [comment]
Both losses are equivalent. It's just a matter of whether `logvar` is being minimized or maximized. As for dividing by `sigma_data`, that's just a constant I put in so you could change `sigma_data`  without affecting training dynamics.
